commands.go

```diff
@@ -3,475 +3,289 @@ package main
-	}
-	if strings.HasPrefix(line, "./banIP") && !isSlack {
-		if !auth(u) {
-			b(devbot, "Not authorized")
-			return
-		}
-		bansMutex.Lock()
-		bans = append(bans, strings.TrimSpace(strings.TrimPrefix(line, "./banIP")))
-		bansMutex.Unlock()
-		saveBansAndUsers()
+	target, ok := findUserByName(u.room, args[0])
+	if !ok {
+		u.system("User not found!")
 		return
 	}
+	target.system(fmt.Sprintf("You have been kicked by %s", u.name))
+	target.close(fmt.Sprintf(red.Paint("%s was kicked by %s"), target.name, u.name))
+	u.system("Kicked!")
+}
 
-	if strings.HasPrefix(line, "./ban") && !isSlack {
-		victim, ok := findUserByName(u.room, strings.TrimSpace(strings.TrimPrefix(line, "./ban")))
-		if !ok {
-			b("", "User not found")
-			return
-		}
-		if !auth(u) {
-			b(devbot, "Not authorized")
-			return
-		}
-		bansMutex.Lock()
-		bans = append(bans, victim.addr)
-		bansMutex.Unlock()
-		saveBansAndUsers()
-		victim.close(victim.name + " has been banned by " + u.name)
+func banCommand(u *user, args []string) {
+	if len(args) == 0 {
+		u.system("Please provide a user to ban!")
 		return
 	}
-	if strings.HasPrefix(line, "./kick") && !isSlack {
-		victim, ok := findUserByName(u.room, strings.TrimSpace(strings.TrimPrefix(line, "./kick")))
-		if !ok {
-			b("", "User not found")
-			return
-		}
-		if !auth(u) {
-			b(devbot, "Not authorized")
-			return
-		}
-		victim.close(victim.name + red.Paint(" has been kicked by ") + u.name)
+	target, ok := findUserByName(u.room, args[0])
+	if !ok {
+		u.system("User not found!")
 		return
 	}
-	if strings.HasPrefix(line, "./color") && !isSlack {
-		rest := strings.TrimSpace(strings.TrimPrefix(line, "./color"))
-		if rest == "which" {
-			b(devbot, "you're using "+u.color)
-			return
-		}
-		if err := u.changeColor(rest); err != nil {
-			b(devbot, err.Error())
-		}
-		return
+	if len(args) > 1 {
+		target.ban(u, strings.Join(args[1:], " "))
+	} else {
+		target.ban(u, "")
 	}
-	if line == "./people" {
-		b("", `
-**Hack Club members**  
-Zach Latta     - Founder of Hack Club  
-Zachary Fogg   - Hack Club Game Designer  
-Matthew        - Hack Club HQ  
-Caleb Denio, Safin Singh, Eleeza A  
-Jubril, Sarthak Mohanty, Anghe,  
-Tommy Pujol, Sam Poder, Rishi Kothari,  
-Amogh Chaubey, Ella Xu, Hugo Hu,  
-Robert Goll, Tanishq Soni, Arash Nur Iman,  
-Temi, Aiden Bai, Ivan Bowman, @epic  
-_Possibly more people_
-
-
-**From my school:**  
-Kiyan, Riya, Georgie  
-Rayed Hamayun, Aarush Kumar
-
-
-**From Twitter:**  
-Ayush Pathak    @ayshptk  
-Bereket         @heybereket  
-Sanketh         @SankethYS  
-Tony Dinh       @tdinh\_me  
-Srushti         @srushtiuniverse  
-Surjith         @surjithctly  
-Arav Narula     @HeyArav  
-Krish Nerkar    @krishnerkar\_  
-Amrit           @astro_shenava  
-Mudrank Gupta   @mudrankgupta  
-Harsh           @harshb__
-
-**And many more have joined!**`)
+	u.system("Banned!")
+}
+
+func idCommand(u *user, args []string) {
+	if len(args) != 1 {
+		u.system(u.id)
```

Comment:
make this check if len is 0

Reply:
Good idea!

Reply:
no problemo, please make args a string and not an []string later on

Reply:
Is that okay @TAG-Epic?

Reply:
I personally dislike this but possibly pass both? Or use the command opts to let the command writer decide

Reply:
Sorry that you dislike it, but I'm convinced it'll be best. For example, if you use the slice way, and then check suppose whether `args[0]` is a valid room. Then if `#foo` is a room, both `./room #foo` and `./room #foo bar qantas` will work. It's better design to make it fail on the second example.

Reply:
So do you agree @TAG-Epic